internal/controller/serverclaim_controller.go

```diff
@@ -61,7 +61,7 @@ func (r *ServerClaimReconciler) delete(ctx context.Context, log logr.Logger, cla
 	log.V(1).Info("Deleting server claim")
 
 	server := &metalv1alpha1.Server{}
-	if err := r.Client.Get(ctx, client.ObjectKey{Name: claim.Spec.ServerRef.Name}, server); err != nil {
+	if err := r.Client.Get(ctx, client.ObjectKey{Name: claim.Spec.ServerRef.Name}, server); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
I would add here an if statement + logging. Something along the line: `log.Info("Server gone")`

Reply:
Sounds good, made changes !!